tuist/tuist
# Fix for missing transitive precompiled static frameworks

Resolves https://github.com/tuist/tuist/issues/3234

### Short description 📝

- The final target that can link products (e.g. Apps, Dynamic Frameworks, Unit Test bundles) was missing transitive precomplied static frameworks
- This was caused by limiting linking of transitive precomplied frameworks to dynamic ones only which isn't necessary
- The changes now allow for all transitive precomplied frameworks to get linked
- Fixtures have been updated to allow testing some of the use cases where this may occur

### Test Plan 🛠

- Verify all CI checks pass
- Manually run

```sh
swift build
swift run tuist generate --path projects/tuist/fixtures/ios_app_with_static_frameworks
```

- Open the generate workspace
- Attempt the compile the `ATests` target and verify it passes
- Manually run

```sh
swift build
swift run tuist dependencies fetch --path projects/tuist/fixtures/app_with_spm_dependencies
swift run tuist generate --path projects/tuist/fixtures/app_with_spm_dependencies
```

- Open the generated workspace
- Attempt to build the App target and verify is passes

### Checklist ✅

- ~In case the PR introduces changes that affect users, the documentation has been updated.~
